engine/execution/computation/computer/computer.go

```diff
@@ -163,53 +156,30 @@ func (e *blockComputer) executeBlock(
 		fvm.WithDerivedBlockData(derivedBlockData))
 	collections := block.Collections()
 
-	res := execution.NewEmptyComputationResult(block)
-
 	var txIndex uint32
 	var err error
-	var wg sync.WaitGroup
-	wg.Add(2) // block commiter and event hasher
-
-	chunksSize := len(collections) + 1 // + 1 system chunk
-	bc := blockCommitter{
-		committer: e.committer,
-		blockSpan: blockSpan,
-		tracer:    e.tracer,
-		metrics:   e.metrics,
-		state:     *block.StartState,
-		views:     make(chan state.View, chunksSize),
-		res:       res,
-	}
-	defer bc.Close()
 
-	eh := eventHasher{
-		tracer:    e.tracer,
-		data:      make(chan flow.EventsList, chunksSize),
-		blockSpan: blockSpan,
-		res:       res,
-	}
-	defer eh.Close()
+	collector := newResultCollector(
+		e.tracer,
+		blockSpan,
+		e.metrics,
+		e.committer,
+		block,
+		len(collections)+1)
```

Comment:
Maybe keep this comment `// + 1 system chunk` just to make it more obvious.

Reply:
This comment stops making sense after #3629